2023-09-07: Add freeze period exception to handbook (#13631)
fleetdm/fleet: handbook/engineering/README.md

## Release process

- [Release freeze period](#release-freeze-period)

> - [Freeze day](#freeze-day)

Reviewer: ```suggestion

```

> - [Merging during freeze](#merging-features-during-freeze)

Reviewer: This will be a broken link, because the heading doesn't match.  I recommend removing the little table of contents lists.  They have great intentions, but we spend far too much time cleaning them up.  It increases the numbers of lines changed in diffs.  It's not worth the maintenance to have these.

Reviewer: ```suggestion

```

Next, create a new GitHub issue using the [Release QA template](https://github.com/fleetdm/fleet/issues/new?assignees=&labels=&projects=&template=smoke-tests.md&title=). Add the release version to the title, and assign the quality assurance members of the [MDM](https://fleetdm.com/handbook/company/development-groups#mdm-group) and [CX](https://fleetdm.com/handbook/company/development-groups#customer-experience-group) product groups.

### Merging during the freeze period

> We merge bug fixes during the freeze period, but we do not merge other code changes. This minimizes code churn and helps ensure a stable release. To merge a bug fix, you must first unfreeze the PR in [Merge Freeze](https://app.mergefreeze.com/installations/3704/branches), and click the "Unfreeze 1 pull request" text link. 

Reviewer 2: ```suggestion
We merge bug fixes and documentation changes during the freeze period, but we do not merge other code changes. This minimizes code churn and helps ensure a stable release. To merge a bug fix, you must first unfreeze the PR in [Merge Freeze](https://app.mergefreeze.com/installations/3704/branches), and click the "Unfreeze 1 pull request" text link. 
```

Small suggestion to add docs here^. 

Then, I think we can get rid of this note below to dedup (note is in the [Freeze day section](https://fleetdm.com/handbook/engineering#freeze-day)):

![Screenshot 2023-08-31 at 2 57 16 PM](https://github.com/fleetdm/fleet/assets/47070608/380579cf-6011-4ecd-960c-5b9084f9be65)

> #### Exceptions

Reviewer: This is a generic heading called "Exceptions", which doesn't make for the clearest URL:

```
fleetdm.com/handbook/engineering#exceptions
```

(like, what kind of exceptions?)

I think this heading can be removed.  In this case, the context of the content is clear based on its "Merging during the freeze period" heading.

Reviewer: ```suggestion

```

It is sometimes necessary to delay the release to allow time to complete partially merged feature work. In these cases, an exception process must be followed before merging during the freeze period. 

> 1. The engineer requesting the feature work merge exception during freeze notifies their EM. 

Reviewer 3: ```suggestion
1. The engineer requesting the feature work merge exception during freeze notifies their Engineering Manager. 
```

> 2. The EM notifies the QA lead for the product group and the [release ritual DRI](https://fleetdm.com/handbook/engineering#rituals). 

Reviewer 3: ```suggestion
2. The Engineering Manager notifies the QA lead for the product group and the [release ritual DRI](https://fleetdm.com/handbook/engineering#rituals). 
```

> 3. The EM, QA lead, and release ritual DRI must all approve the feature work PR before it is unfrozen and merged.

Reviewer 3: ```suggestion
3. The Engineering Manager, QA lead, and [release ritual DRI](#rituals) must all approve the feature work PR before it is unfrozen and merged.
```

## Rituals

> The following rituals are engaged in by the directly responsible individual (f) and at the frequency specified for the ritual.

Reviewer: "f"?

Author: F is for fat fingered. Fixed.
